File: app/src/main/java/net/simplifiedcoding/simplegame/Player.java
package net.simplifiedcoding.simplegame;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Rect;

public class Player {
    private Rect detectCollision;
    private Bitmap bitmap;

    private int x;
    private int y;

    private int lift = 0;
    private boolean lifting;

    private int maxY;
    private int minY;

    public Player(Context context, int screenX, int screenY) {
        x = 75;
        y = 50;
        lift = 1;

        bitmap = BitmapFactory.decodeResource(context.getResources(), R.drawable.player);
        maxY = screenY - bitmap.getHeight();
        minY = 0;
        lifting = false;

        detectCollision = new Rect(x, y, bitmap.getWidth(), bitmap.getHeight());
    }
    public void setLifting() {
        lifting = true;
    }
    public void stopLifting() {
        lifting = false;
    }
    public void update() {
        if (lifting) {
            lift = 10;
        } else {
            lift = -10;
        }

        y -= lift;

        if (y < minY) {
            y = minY;
        }
        if (y > maxY) {
            y = maxY;
        }
        detectCollision.left = x;
        detectCollision.top = y;
        detectCollision.right = x + bitmap.getWidth();
        detectCollision.bottom = y + bitmap.getHeight();
    }
    public Rect getDetectCollision() {
        return detectCollision;
    }
    public Bitmap getBitmap() {
        return bitmap;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

}

File: app/src/main/java/net/simplifiedcoding/simplegame/Player.kt
package net.simplifiedcoding.simplegame

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Rect

class Player(context: Context, screenX: Int, screenY: Int) {
    @JvmField
    val detectCollision: Rect
    @JvmField
    val bitmap: Bitmap
    @JvmField
    val x = 75
    var y = 50
        private set
    private var lift = 0
    private var lifting: Boolean
    private val maxY: Int
    private val minY: Int
    fun setLifting() {
        lifting = true
    }

    fun stopLifting() {
        lifting = false
    }

    fun update() {
        lift = if (lifting) {
            10
        } else {
            -10
        }
        y -= lift
        if (y < minY) {
            y = minY
        }
        if (y > maxY) {
            y = maxY
        }
        detectCollision.left = x
        detectCollision.top = y
        detectCollision.right = x + bitmap.width
        detectCollision.bottom = y + bitmap.height
    }

    init {
        lift = 1
        bitmap = BitmapFactory.decodeResource(context.resources, R.drawable.player)
        maxY = screenY - bitmap.height
        minY = 0
        lifting = false
        detectCollision = Rect(x, y, bitmap.width, bitmap.height)
    }
}